### PR TITLE
Fix keep-alive config not working

### PR DESCRIPTION
**Pull request by riyafa**
## Automation tests
 - Unit tests 
   > Added in transport-http

### CODE REVIEW

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 1]
```diff
@@ -588,28 +588,16 @@ public static BStruct createNewEntity(Context context, BStruct struct) {
      * @param context         ballerina context.
      * @param outboundMessage transport message.
      */
-    public static void setKeepAliveAndCompressionHeaders(Context context, HTTPCarbonMessage outboundMessage) {
+    public static void setCompressionHeaders(Context context, HTTPCarbonMessage outboundMessage) {
```
Comment (VijithaEkanayake) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Don't we need to change the doc comment with respect to the change? As this method only sets compression header now.

Comment (riyafa) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Fixed

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 2]
```diff
@@ -719,16 +708,17 @@ private static void extractBasicConfig(Annotation configInfo, Set<ListenerConfig
...
 
-    public static ChunkConfig getChunkConfig(String chunking) {
-        ChunkConfig chunkConfig;
-        if (HttpConstants.CHUNKING_AUTO.equalsIgnoreCase(chunking)) {
-            chunkConfig = ChunkConfig.AUTO;
-        } else if (HttpConstants.CHUNKING_ALWAYS.equalsIgnoreCase(chunking)) {
-            chunkConfig = ChunkConfig.ALWAYS;
-        } else if (HttpConstants.CHUNKING_NEVER.equalsIgnoreCase(chunking)) {
-            chunkConfig = ChunkConfig.NEVER;
+    public static ChunkKeepAliveConfig getKeepAliveChunkConfig(String keepAliveChunkConfig) {
```
Comment (lankavitharana) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: can't we use a switch case here?

Comment (pubudu91) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Please refer to my comment on mixing keep-alive configs and chunking configs in [https://github.com/wso2/transport-http/pull/60](https://github.com/wso2/transport-http/pull/60)

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 3]
```diff
@@ -719,16 +708,17 @@ private static void extractBasicConfig(Annotation configInfo, Set<ListenerConfig
...
+        if (HttpConstants.CHUNKING_KEEP_ALIVE_AUTO.equalsIgnoreCase(keepAliveChunkConfig)) {
+            chunkConfig = ChunkKeepAliveConfig.AUTO;
+        } else if (HttpConstants.CHUNKING_KEEP_ALIVE_ALWAYS.equalsIgnoreCase(keepAliveChunkConfig)) {
+            chunkConfig = ChunkKeepAliveConfig.ALWAYS;
+        } else if (HttpConstants.CHUNKING_KEEP_ALIVE_NEVER.equalsIgnoreCase(keepAliveChunkConfig)) {
+            chunkConfig = ChunkKeepAliveConfig.NEVER;
         } else {
-            throw new BallerinaConnectorException("Invalid configuration found for Transfer-Encoding : " + chunking);
+            throw new BallerinaConnectorException(
+                    "Invalid configuration found for Transfer-Encoding : " + keepAliveChunkConfig);
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: This is a wrong exception in the case of keep-alive configurations.

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/test/StartService.java [hunk 4]
```diff
@@ -198,80 +199,66 @@ private String getServiceURL(ServiceInfo service) {
...
             }
-            
+
             if (keepAliveAttrVal != null) {
-                listenerConfiguration.setKeepAlive(keepAliveAttrVal.getBooleanValue());
+                listenerConfiguration.setKeepAliveConfig(
+                        HttpUtil.getChunkKeepAliveConfig(keepAliveAttrVal.getStringValue()));
             } else {
-                listenerConfiguration.setKeepAlive(Boolean.TRUE);
+                listenerConfiguration.setKeepAliveConfig(ChunkKeepAliveConfig.AUTO);
```
Comment (chamil321) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/test/StartService.java]: ChunkKeepAliveConfig is very confusing and hard to understand. Shall we use separate config for keepAlive?

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal [hunk 5]
```diff
@@ -54,7 +55,7 @@ public annotation configuration attach service<> {
     string[] allowHeaders;
     int maxAge;
     string[] exposeHeaders;
-    boolean keepAlive;
+    string keepAlive;
```
Comment (irunika) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Why can't we use enum instead of String here?

Comment (irunika) [reply stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Please check https://github.com/ballerina-lang/ballerina/blob/master/tests/ballerina-test/src/test/resources/test-src/enums/enums.bal

Comment (riyafa) [reply stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: then should the enum be defined somewhere so the user can use it?

Comment (irunika) [reply stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Yes. It should come as a part of http package which will look like ```http:KeepAlive.AUTO```

Comment (riyafa) [reply stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Tracked by this issue for both chunking and keep-alive config https://github.com/ballerina-lang/ballerina/issues/4961

**Pull request merged**